### PR TITLE
[WIP] SCTP: RTCSctpTransport, RTCDtlsTransport, and RTCIceTransport

### PR DESCRIPTION
**Pull request by trivigy**
#83 This is where I started the entire proposal and this is the main PR for it. In summary I am trying to eliminate networkManager and instead break down the different functionalities into their respective protocol layers. SCTP vs. DTLS vs. ICE/UDP vs. RTP

### CODE REVIEW

File: internal/sctp/chunk_init.go [hunk 1]
```diff
@@ -11,12 +11,12 @@ See chunkInitCommon for the fixed headers
 
 Variable Parameters                  Status     Type Value
 -------------------------------------------------------------
-IPv4 IP (Note 1)               Optional    5
-IPv6 IP (Note 1)               Optional    6
+IPv4 IP (Note 1)               		Optional    5
```
[the remark this answers is not in the document]
Comment (trivigy) [reply internal/sctp/chunk_init.go]: Yeah. These are tooling changes. Will fix.

File: internal/sctp/association.go [hunk 2]
```diff
@@ -75,20 +75,20 @@ type Association struct {
 	peerVerificationTag uint32
 	myVerificationTag   uint32
 	state               AssociationState
-	//peerTransportList
-	//primaryPath
-	//overallErrorCount
-	//overallErrorThreshold
-	//peerReceiverWindow (peerRwnd)
+	// peerTransportList
```
[the remark this answers is not in the document]
Comment (trivigy) [reply internal/sctp/association.go]: Yeah. These are tooling changes. Will fix.

File: internal/sctp/association.go [hunk 3]
```diff
@@ -58,7 +58,7 @@ func (a AssociationState) String() string {
 // Verification: in the INIT or INIT ACK chunk.
 // Tag         :
 //
-// My          : Tag expected in every inbound packet and sent in the
+// My          : Tag expected in every reader packet and sent in the
```
[the remark this answers is not in the document]
Comment (trivigy) [reply internal/sctp/association.go]: Yeah. These are tooling changes. Will fix.

**Comment by Sean-Der** [comment]
DTLS 1.2 was implemented out of tree https://github.com/pions/dtls

ICE refactor done here https://github.com/pions/webrtc/pull/224

SCTP hasn't been done yet, but will probably be moved to `github.com/pions/sctp` when everything else lands

**Pull request closed without merging**